In [2]:
from pathlib import Path
import json

html_cache_dir = Path("../data/processed/html_cache")
lang_data_path = Path("../data/processed/language_data.json")

# Build language -> class_id lookup
lang_to_class = {}
if lang_data_path.exists():
    with open(lang_data_path, "r", encoding="utf-8") as f:
        lang_data = json.load(f)
    for class_id, langs in lang_data["lang_classes"].items():
        for lang in langs:
            lang_to_class[lang] = int(class_id)
    print(f"Loaded {len(lang_to_class)} language->class mappings\n")
else:
    print(f"Warning: language_data.json not found at {lang_data_path}\n")

files_with_languages = []

if html_cache_dir.exists():
    json_files = sorted(html_cache_dir.glob("*.json"))
    print(f"Found {len(json_files)} JSON files in html_cache/\n")

    for json_file in json_files:
        paper_id = json_file.stem
        with open(json_file, "r", encoding="utf-8") as f:
            data = json.load(f)

        paper_sections = []
        for section_name, section_data in data.items():
            detected = section_data.get("detected", [])
            filtered = sorted(set(lang for lang in detected if lang_to_class.get(lang, 5) < 5))
            if filtered:
                paper_sections.append({"section": section_name, "languages": filtered})
                files_with_languages.append({
                    "paper_id": paper_id,
                    "file": str(json_file),
                    "section": section_name,
                    "languages": filtered,
                })

        if paper_sections:
            print(f"Paper ID:  {paper_id}")
            print(f"File:      {json_file}")
            for s in paper_sections:
                print(f"  {s['section']}: {len(s['languages'])} unique — {s['languages']}")
            print()

    print(f"\n=== SUMMARY ===")
    print(f"Total sections with detected languages (class 0–4): {len(files_with_languages)}")
else:
    print(f"Directory not found: {html_cache_dir}")

Loaded 6418 language->class mappings

Found 319 JSON files in html_cache/

Paper ID:  2605.17710v1
File:      ../data/processed/html_cache/2605.17710v1.json
  1Introduction: 2 unique — ['Hausa', 'Igbo']
  3Methodology: 3 unique — ['Hausa', 'Igbo', 'Yoruba']
  3.1Dataset curation: 3 unique — ['Hausa', 'Igbo', 'Yoruba']
  Audio processing.: 2 unique — ['Hausa', 'Igbo']
  N-gram language models.: 3 unique — ['Hausa', 'Igbo', 'Yoruba']
  4Experiments: 2 unique — ['Hausa', 'Igbo']
  4.1Average teachers produce good students: 1 unique — ['Igbo']
  4.3Evaluating robustness on conversational speech: 1 unique — ['Igbo']
  4.4Performance gap still exists in diacritical mark prediction: 1 unique — ['Igbo']
  4.5SBPN models are strong language identifiers: 2 unique — ['Hausa', 'Igbo']
  4.6Other experiment details: 2 unique — ['Hausa', 'Igbo']
  Evaluation set and metrics: 2 unique — ['Hausa', 'Igbo']
  5Future Directions: 2 unique — ['Ghanaian Pidgin English', 'Igbo']
  5.2Other research directio

Loaded 6418 language->class mappings

Found 49 JSON files in html_cache/

Paper ID:  2605.22099v1
File:      ../data/processed/html_cache/2605.22099v1.json
  1Introduction: 2 unique — ['Khmer', 'Latin']
  3Methodology: 2 unique — ['Jina', 'Khmer']
  3.1Dataset: 1 unique — ['Khmer']
  3.2Retriever Models: 2 unique — ['Jina', 'Khmer']
  3.4Evaluation Metrics: 1 unique — ['Jina']
  4Results and Discussion: 2 unique — ['Jina', 'Khmer']
  4.1Retriever Evaluation: 1 unique — ['Jina']
  4.3Discussion on Metrics: 1 unique — ['Khmer']
  Answer Relevance (𝐴𝑛𝑠𝑅𝑒𝑙\mathit{AnsRel}).: 1 unique — ['Khmer']
  4.4Retriever Model Selection: 2 unique — ['Jina', 'Khmer']
  4.5Language Focus vs. General Multilingual Models: 1 unique — ['Khmer']
  4.6Limitations: 1 unique — ['Khmer']
  5Conclusion and Future Work: 3 unique — ['Jina', 'Khmer', 'Latin']

Paper ID:  2605.22137v1
File:      ../data/processed/html_cache/2605.22137v1.json
  4Experimental Setup: 8 unique — ['Amharic', 'Assamese', 'Azerbaijani', 'Greek', 'Hausa', 'Indonesian', 'Korean', 'Persian']
  4.1Base Model and Target Languages: 8 unique — ['Amharic', 'Assamese', 'Azerbaijani', 'Greek', 'Hausa', 'Indonesian', 'Korean', 'Persian']
  5Results: 2 unique — ['Assamese', 'Hausa']
  5.1Main Results: 2 unique — ['Assamese', 'Hausa']

Paper ID:  2605.22170v1
File:      ../data/processed/html_cache/2605.22170v1.json
  2Methodology: 1 unique — ['Gan']
  2.2The multimodal large language model under study: TheSpiritLMmodel: 1 unique — ['Gan']

Paper ID:  2605.22202v1
File:      ../data/processed/html_cache/2605.22202v1.json
  Appendix AMethodological Details: 1 unique — ['Jina']
  A.2Embedding models and Prompting: 1 unique — ['Jina']
  A.2.2Embedding Model Licenses and Citations: 1 unique — ['Jina']
  Appendix BComplementary Results: 4 unique — ['Finnish', 'Greek', 'Ik', 'Ukrainian']
  B.2Neighborhood Retention for Suboptimal Number of Neighbors: 2 unique — ['Finnish', 'Greek']
  B.3Stability of ICA: 3 unique — ['Finnish', 'Ik', 'Ukrainian']
  B.4Interpretability through ICA: 1 unique — ['Finnish']

Paper ID:  2605.22203v1
File:      ../data/processed/html_cache/2605.22203v1.json
  1Introduction: 1 unique — ['Khmer']
  2.2Sentence-Based and Language-Aware Chunking: 2 unique — ['Khmer', 'Vietnamese']
  2.3LLM-Based Document Segmentation: 1 unique — ['Khmer']
  2.4Embedding Models for Multilingual Retrieval: 1 unique — ['Khmer']
  2.5Low-Resource Language Processing: 2 unique — ['Khmer', 'Latin']
  3Methodology: 1 unique — ['Khmer']
  3.2Chunking Strategies: 1 unique — ['Khmer']
  3.2.1Embedding Generation: 1 unique — ['Khmer']
  3.3Chunking Strategy Examples: 1 unique — ['Khmer']
  4Evaluation Metrics: 2 unique — ['Khmer', 'Latin']
  4.3Khmer Linguistic Fidelity: 2 unique — ['Khmer', 'Latin']
  4.3.1Khmer Coverage: 2 unique — ['Khmer', 'Latin']
  4.3.2Khmer Intersection-over-Union (KIoU): 1 unique — ['Khmer']
  5Evaluation Results: 1 unique — ['Khmer']
  5.2Retrieval Performance Comparison: 1 unique — ['Khmer']
  5.3Qualitative Analysis: 1 unique — ['Khmer']
  6Discussion: 3 unique — ['Khmer', 'Lao', 'Thai']
  6.1Interpretation of Key Findings: 1 unique — ['Khmer']
  6.2Trade-offs and Strategy Selection Guidelines: 1 unique — ['Khmer']
  6.3Implications for Low-Resource Language Processing: 3 unique — ['Khmer', 'Lao', 'Thai']
  6.4Limitations: 1 unique — ['Khmer']
  7Conclusion and Future Work: 1 unique — ['Khmer']

Paper ID:  2605.22217v1
File:      ../data/processed/html_cache/2605.22217v1.json
  3Training Collapse: Empirical Mechanisms: 1 unique — ['Kwa']
  3.2The Grounded Proposer Paradox: 1 unique — ['Kwa']
  Mechanism.: 1 unique — ['Kwa']
  5.2Reward Hacking and Goodhart’s Law: 1 unique — ['Kwa']

Paper ID:  2605.22247v1
File:      ../data/processed/html_cache/2605.22247v1.json
  2.1Idioms as a Core Challenge in NLP: 1 unique — ['Epie']

Paper ID:  2605.22258v1
File:      ../data/processed/html_cache/2605.22258v1.json
  2.2Chinese Toxicity Detection: 1 unique — ['Pinyin']

Paper ID:  2605.22355v1
File:      ../data/processed/html_cache/2605.22355v1.json
  4Benchmark Tasks: 1 unique — ['Mape']
  4.1Task Definitions: 1 unique — ['Mape']
  4.2Evaluation Metrics: 1 unique — ['Mape']
  Numeric Field Accuracy.: 1 unique — ['Mape']
  5Experiments: 1 unique — ['Mape']
  5.2Benchmark Results: 1 unique — ['Mape']
  Comparison with general-purpose LLMs.: 1 unique — ['Mape']
  Main results.: 1 unique — ['Mape']
  Data scaling.: 1 unique — ['Mape']
  GPS-only ablation.: 1 unique — ['Mape']
  Appendix DEvaluation Metrics: 1 unique — ['Mape']
  D.4Numeric Field Accuracy: 1 unique — ['Mape']
  Appendix FAdditional Experiments: 1 unique — ['Mape']
  F.2Single-City vs. Multi-City CPT: 1 unique — ['Mape']
  F.3Effect of Continual Pre-Training: 1 unique — ['Mape']
  F.5Comparison with Tool-Augmented LLMs: 1 unique — ['Mape']

Paper ID:  2605.22356v1
File:      ../data/processed/html_cache/2605.22356v1.json
  2Methodology: 1 unique — ['Dass']
  2.4Evaluation Framework: 1 unique — ['Dass']
  2.4.2Quantitative analysis: 1 unique — ['Dass']
  3Results: 1 unique — ['Dass']
  3.3Symptom Specificity and Discriminant Validity: 1 unique — ['Dass']

Paper ID:  2605.22391v1
File:      ../data/processed/html_cache/2605.22391v1.json
  1Introduction: 5 unique — ['Indonesian', 'Pole', 'Russian', 'Turkish', 'Vietnamese']
  2Methods: 6 unique — ['Hungarian', 'Indonesian', 'Pole', 'Russian', 'Turkish', 'Vietnamese']
  2.1Corpus: 4 unique — ['Indonesian', 'Russian', 'Turkish', 'Vietnamese']
  2.5Evaluation: 2 unique — ['Hungarian', 'Pole']
  Emergent geometry.: 2 unique — ['Hungarian', 'Pole']
  3Geometry: 2 unique — ['Latin', 'Pole']
  3.1Isotropy and food-group structure: 1 unique — ['Latin']
  3.3Emergent factors and modes: 2 unique — ['Latin', 'Pole']
  4Transformations: 4 unique — ['Indonesian', 'Italian', 'Latin', 'Pole']
  4.1Pairings: 2 unique — ['Italian', 'Pole']
  Mode-membership pairings.: 1 unique — ['Pole']
  4.2Direction arithmetic: 3 unique — ['Indonesian', 'Latin', 'Pole']
  SLERP toward supervised directions.: 2 unique — ['Latin', 'Pole']
  SLERP toward emergent mode poles.: 1 unique — ['Indonesian']
  5Discussion: 2 unique — ['Latin', 'Pole']
  5.2From recommendation to navigation: 1 unique — ['Pole']
  5.3Limitations: 1 unique — ['Latin']
  Corpus imbalance.: 1 unique — ['Latin']

Paper ID:  2605.22465v1
File:      ../data/processed/html_cache/2605.22465v1.json
  1Introduction and Theoretical Background: 1 unique — ['Elu']
  5Discussion & Future Directions: 1 unique — ['Elu']

Paper ID:  2605.22536v1
File:      ../data/processed/html_cache/2605.22536v1.json
  4Experiments: 1 unique — ['Deg']
  4.3Degradation-wise Correlation Analysis: 1 unique — ['Deg']
  Metric Design: 1 unique — ['Deg']

Paper ID:  2605.22542v1
File:      ../data/processed/html_cache/2605.22542v1.json
  1Introduction: 1 unique — ['Crow']
  Frame Semantics.: 1 unique — ['Crow']
  5Experiments: 2 unique — ['Fur', 'Pin']
  5.2.2Results: 2 unique — ['Fur', 'Pin']
  Dimension-specific findings.: 2 unique — ['Fur', 'Pin']
  Appendix: 1 unique — ['Crow']
  A COCA-Scenes Keywords: 1 unique — ['Crow']

Paper ID:  2605.22544v1
File:      ../data/processed/html_cache/2605.22544v1.json
  1Introduction and Background: 1 unique — ['Jina']

Paper ID:  2605.22616v1
File:      ../data/processed/html_cache/2605.22616v1.json
  1Study 1: Sensorimotor and embodiment ratings for 3,000 Chinese lexicalized concepts: 1 unique — ['Italian']
  1.4Results and discussion: 1 unique — ['Italian']
  1.4.2Descriptive statistics: 1 unique — ['Italian']
  Dominance and exclusivity scores: 1 unique — ['Italian']
  1.4.3Correlation analyses: 1 unique — ['Italian']
  Appendix: 1 unique — ['Pinyin']
  4.2Mini-test on Chinese Language: 1 unique — ['Pinyin']
  Sensorimotor survey mini-test: 1 unique — ['Pinyin']
  Embodiment survey mini-test: 1 unique — ['Pinyin']

Paper ID:  2605.22660v1
File:      ../data/processed/html_cache/2605.22660v1.json
  1.Introduction: 1 unique — ['Polish']
  3.Methods: 1 unique — ['Polish']
  3.3.Translation Prompts: 1 unique — ['Polish']
  3.4.Validation Methods: 1 unique — ['Polish']
  4.Results: 1 unique — ['Polish']
  4.2.Embedding Similarity and CKA: 1 unique — ['Polish']
  4.4.Full Fine-Tuning Validation: 1 unique — ['Polish']
  5.Discussion: 1 unique — ['Polish']
  6.Conclusion: 1 unique — ['Polish']

Paper ID:  2605.22715v1
File:      ../data/processed/html_cache/2605.22715v1.json
  1Introduction: 1 unique — ['Mo']

Paper ID:  2605.22732v1
File:      ../data/processed/html_cache/2605.22732v1.json
  2.1Speech Emotion Recognition: 1 unique — ['Wa']

Paper ID:  2605.22769v1
File:      ../data/processed/html_cache/2605.22769v1.json
  1Introduction: 1 unique — ['Greek']

Paper ID:  2605.22785v1
File:      ../data/processed/html_cache/2605.22785v1.json
  1Introduction: 3 unique — ['Hindi', 'Russian', 'Turkish']
  2Experimental Setup: 4 unique — ['Hindi', 'Latin', 'Russian', 'Turkish']
  2.1News Sources and Coverage: 4 unique — ['Hindi', 'Latin', 'Russian', 'Turkish']
  3Main Results: 3 unique — ['Hindi', 'Russian', 'Turkish']
  3.2Regional Disparities: The Hindi Gap: 3 unique — ['Hindi', 'Russian', 'Turkish']
  4Analysis: 3 unique — ['Hindi', 'Russian', 'Turkish']
  4.1Citation Behavior and Source Attribution: 3 unique — ['Hindi', 'Russian', 'Turkish']
  4.1.1Citation Rates & Dissociation from Accuracy: 2 unique — ['Hindi', 'Turkish']
  4.1.3English Dominance in Non-English Contexts: 3 unique — ['Hindi', 'Russian', 'Turkish']
  4.2Error Taxonomy:WhyModels Fail: 2 unique — ['Hindi', 'Russian']
  4.3The Role of Web Search: Ablation Study: 1 unique — ['Hindi']
  5Discussion: 1 unique — ['Hindi']
  5.1Limitations: 1 unique — ['Hindi']
  6Conclusion: 1 unique — ['Hindi']
  Acknowledgment: 1 unique — ['Batu']
  Appendix DModel×\timesRegion Interactions: 3 unique — ['Hindi', 'Russian', 'Turkish']
  Appendix EMultiple-Choice vs. Free-Response Validation: Extended Results: 2 unique — ['Hindi', 'Russian']
  F.1Where multiple-choice over-states accuracy: gap heatmap: 1 unique — ['Hindi']
  Appendix HAdditional Methodological Notes: 3 unique — ['Hindi', 'Russian', 'Turkish']
  Time-zone targeting.: 3 unique — ['Hindi', 'Russian', 'Turkish']
  Appendix ISelected Model Responses: 6 unique — ['Aka', 'Hindi', 'Irish', 'Russian', 'Turkish', 'Yale']
  I.1Correctly answered questions: 1 unique — ['Irish']
  I.2Incorrectly answered questions: 4 unique — ['Hindi', 'Russian', 'Turkish', 'Yale']
  I.3Adversarially-perturbed questions: 1 unique — ['Aka']

Paper ID:  2605.22817v1
File:      ../data/processed/html_cache/2605.22817v1.json
  3Method: Vector Policy Optimization: 1 unique — ['Puri']
  3.1Multi-Answer Chains as In-Context Exploration: 1 unique — ['Puri']


=== SUMMARY ===
Total sections with detected languages (class 0–4): 153

In [1]:
import sys, json
sys.path.insert(0, "..")

from langtrend.text_cleaning import clean_paper_text_for_language_screening, detect_languages_in_text

# Load language data
with open("../data/processed/language_data.json", encoding="utf-8") as f:
    _lang_data = json.load(f)
lang_classes = {int(k): set(v) for k, v in _lang_data["lang_classes"].items()}
languages_to_ignore = set(_lang_data["languages_to_ignore"])
print(f"Loaded {sum(len(v) for v in lang_classes.values())} languages across {len(lang_classes)} classes\n")

# --- paste your sample text here ---
sample_text = r"""
\\displaystyle\\sigma_{ij}=\\left|\\frac{\\text{Cov}(s_{i},s^{\\prime}_{j})}{\\sqrt{\\frac{1}{d}\\sum_{k}s_{ik}^{2}}\\sqrt{\\frac{1}{d}\\sum_{k}{s^{\\prime}}_{jk}^{2}}}
"""

cleaned_blocks, step_matches = clean_paper_text_for_language_screening(sample_text)
detected = detect_languages_in_text(cleaned_blocks, lang_classes, languages_to_ignore)

print("=== Cleaned blocks ===")
for i, block in enumerate(cleaned_blocks):
    print(f"  [{i}] {block}")

print("\n=== Removed by step ===")
for step, removed in step_matches.items():
    print(f"  {step}: {removed}")

print(f"\n=== Detected languages ===")
print(detected)

Loaded 6419 languages across 6 classes

=== Cleaned blocks ===
  [0] displaystyle sigma | (s s ) sqrt d sum s sqrt d sum s

=== Removed by step ===
  math_commands: ['\\left', '\\frac', '\\text{Cov}', '\\frac{1}', '\\frac{1}']
  subscripts_superscripts: ['_{ij}', '_{i}', '^{\\\\prime}', '_{j}', '_{k}', '_{ik}', '^{2}', '_{k}', '^{\\\\prime}', '_{jk}']
  replaced_chars: Counter({'\\': 17, '{': 7, '}': 7, '=': 1, ',': 1})

=== Detected languages ===
[]
